x/nexus/keeper/general_message.go

```diff
@@ -113,12 +105,12 @@ func (k Keeper) SetMessageExecuted(ctx sdk.Context, messageID exported.MessageID
 		return fmt.Errorf("general message %s not found", messageID.String())
 	}
 
-	if !(m.Is(exported.Sent) || m.Is(exported.Approved)) {
+	if !(m.Is(exported.Sent)) {
 		return fmt.Errorf("general message is not sent or approved")
 	}
 
-	if m.Is(exported.Approved) {
-		k.deleteApprovedMessage(ctx, m)
+	if m.Is(exported.Sent) {
```

Comment:
this `if` is now redundant, you always want to delete it now since it'll be in the sent status